starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -82,7 +83,11 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    Query query = new Query(Criteria.where("id").is(id));
+    Inventory curr = this.mongoTemplate.findAndRemove(query, Inventory.class);
+    if (curr == null) {
+      return Optional.empty();
+    }
+    return Optional.of(curr);
```

Comment:
You can even shorten this a little bit more by returning `Optional.ofNullable(findAndRemove(query, Inventory.class))`, which checks for the same case that you're checking for here.
As the name suggests, `Optional.ofNullable()` allows for a null value. Overall, though, the functionality is good!